packages/fast-components-react-base/src/checkbox/checkbox.tsx

```diff
@@ -5,51 +5,52 @@ import { CheckboxHTMLTags, CheckboxProps, ICheckboxHandledProps, ICheckboxManage
 import { ICheckboxClassNameContract, IManagedClasses } from "@microsoft/fast-components-class-name-contracts-base";
 import { get } from "lodash-es";
 
+/**
+ * Checkbox state interface
+ */
+export interface ICheckboxState {
+    checked: boolean;
+}
+
 /* tslint:disable-next-line */
-class Checkbox extends Foundation<ICheckboxHandledProps & ICheckboxManagedClasses, ICheckboxUnhandledProps, {}> {
+class Checkbox extends Foundation<ICheckboxHandledProps & ICheckboxManagedClasses, ICheckboxUnhandledProps, ICheckboxState> {
     protected handledProps: HandledProps<ICheckboxHandledProps & IManagedClasses<ICheckboxClassNameContract>> = {
         managedClasses: void 0,
         text: void 0,
         checked: void 0,
         indeterminate: void 0,
-        onChange: void 0,
         disabled: void 0,
         tag: void 0
     };
 
     /**
-     * Stores HTML tag for use in render
+     * Provides reference to input
      */
-    private get tag(): string {
-        return CheckboxHTMLTags[this.props.tag] || CheckboxHTMLTags.label;
-    }
-
     private inputRef: React.RefObject<HTMLInputElement>;
-        constructor(props: CheckboxProps) {
-            super(props);
-
-            this.inputRef = React.createRef();
-        }
 
     /**
-     * Apply indeterminate state to items that are indeterminate.
-     * This method should be called after render because it relies on element references.
+     * Define constructor
      */
-    public applyIndeterminateState(): any {
-         if (this.props.indeterminate !== undefined && this.inputRef.current) {
-            this.inputRef.current.indeterminate = this.props.indeterminate;
-        }
+    constructor(props: CheckboxProps) {
+        super(props);
+
+        this.state = {
+            checked: this.props.checked
```

Comment:
The prop `checked` is optional but the state checked is required, therefore its possible that state checked can be undefined, can you make sure that this is always a boolean?

Reply:
Roger